core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3221,11 +3221,12 @@ private[spark] object Utils extends Logging {
    * Return the median number of a long array
    *
    * @param sizes
+   * @param alreadySorted
    * @return
    */
-  def median(sizes: Array[Long]): Long = {
+  def median(sizes: Array[Long], alreadySorted: Boolean = false): Long = {
```

Comment:
What about always assuming the `sizes` are sorted (mentioning this in the description and reflecting in the parameter name ie. `sortedSizes` or in the name of the method..)?

Reply:
Hi, @attilapiros , in my opinion,  `median` method should support unsorted input array. But if we change the parameter name to `sortedSizes`, that will be a little confusing?

Reply:
Ok, but in that case I won't give a default value for the `alreadySorted`, let's require the client to define it.

Reply:
Agree, I have updated `median` method.